src/core/input_descriptor.rs

```diff
@@ -834,3 +822,66 @@ pub enum ConstraintsLimitDisclosure {
     Required,
     Preferred,
 }
+
+/// The [RequestedField] type is non-normative and is not part of the
+/// core OID4VP specification. However, it is provided as a helper function
+/// for returning requested fields parsed from a given credential that
+/// correspond to the input descriptor constraint fields that are requested.
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct RequestedField {
+    /// A unique ID for the requested field
+    pub id: Uuid,
+    pub name: String,
+    pub required: bool,
+    pub retained: bool,
+    pub purpose: Option<String>,
+    pub constraint_field_id: Option<String>,
+    // the `raw_field` represents the actual field
+    // being selected by the input descriptor JSON path
+    // selector.
+    pub raw_fields: Option<serde_json::Value>,
+}
+
+impl RequestedField {
+    /// Construct a new requested field given the required parameters. This method is exposed as
+    /// public, however, it is likely that the `from_definition` method will be used to construct
+    /// requested fields from a presentation definition.
+    ///
+    /// See [RequestedField::from_definition] to return a vector of requested fields
+    /// according to a presentation definition.
+    pub fn new(
+        name: String,
+        required: bool,
+        retained: bool,
+        purpose: Option<String>,
+        constraint_field_id: Option<String>,
+        raw_fields: Option<serde_json::Value>,
+    ) -> Self {
+        Self {
+            id: Uuid::new_v4(),
+            name,
+            required,
+            retained,
+            purpose,
+            constraint_field_id,
+            raw_fields,
+        }
+    }
+
+    /// Return the unique ID for the request field.
+    pub fn id(&self) -> Uuid {
+        self.id
+    }
+
+    /// Return the constraint field id this requested field belongs to,
+    /// if it exists.
+    pub fn constraint_field_id(&self) -> Option<String> {
+        self.constraint_field_id.clone()
+    }
+
+    /// Return the raw values of the credential matching the
+    /// constraint fields filter.
+    pub fn raw_fields(&self) -> Option<&serde_json::Value> {
+        self.raw_fields.as_ref()
+    }
+}
```

Comment:
These fields are all public, so the getters and the constructor are not needed. In particular, constructing via `RequestedField { ... }` is preferred over a constructor, particular as it is easy in this case to get the parameters in the wrong order.
```suggestion
```